agent/structs/config_entry.go

```diff
@@ -25,6 +26,11 @@ type ConfigEntry interface {
 	Normalize() error
 	Validate() error
 
+	// VerifyReadACL and VerifyWriteACL return whether or not the given Authorizer
+	// has permission to read or write to the config entry, respectively.
+	VerifyReadACL(acl.Authorizer) bool
+	VerifyWriteACL(acl.Authorizer) bool
```

Comment:
How do you feel about names like `CanRead(), CanWrite()`. I think this looks great but just the conditionals ended up not reading super cleanly to me:

```go
if rule != nil && !args.Entry.VerifyWriteACL(rule) {
```

Is kinda hard to parse and figure out the negations and then assume that "Verify" returns true on failure etc. Whereas this seems slightly clearer as it's an assertion:

```go
if rule != nil && !args.Entry.CanWrite(rule) {
```

It's minor since we shouldn't need to call these in too many other places now but just a thought.